src/gluonnlp/optimizer/bert_adam.py

```diff
@@ -88,48 +91,96 @@ def update(self, index, weight, grad, state):
 
     def update_multi_precision(self, index, weight, grad, state):
         """multi-precision update function"""
-        use_multi_precision = self.multi_precision and weight.dtype == numpy.float16
+        use_multi_precision = self.multi_precision and weight[0].dtype == numpy.float16
         self._update_impl(index, weight, grad, state,
                           multi_precision=use_multi_precision)
 
     def _update_impl(self, indices, weight, grad, state, multi_precision=False):
         """update function"""
         try:
-            from mxnet.ndarray.contrib import adamw_update
+            from mxnet.ndarray.contrib import adamw_update, multi_adamw_update
         except ImportError:
             raise ImportError('Failed to import nd.contrib.adamw_update from MXNet. '
                               'BERTAdam optimizer requires mxnet>=1.5.0b20190220. '
                               'Please upgrade your MXNet version. For example: '
                               'pip install mxnet-cu90 --pre. Otherwise, please consider '
                               'Adam optimizer with different hyper-parameters.')
+        if multi_precision:
+            try:
+                from mxnet.ndarray.contrib import mp_adamw_update, multi_mp_adamw_update
+            except ImportError:
+                raise ImportError('Failed to import '
```

Comment:
We currently target MXNet 1.6 for the upcoming gluonnlp release. You can remove these checks.

Reply:
+1. we can also remove the checks for adamw_update, multi_adamw_update